modules/assim.batch/R/pda.emulator.R

```diff
@@ -194,13 +321,33 @@ pda.emulator <- function(settings, params.id=NULL, param.names=NULL, prior.id=NU
       
       ## Convert probabilities back to parameter values
       for(i in 1:n.param) {
-        m[,i] <- eval(prior.fn$qprior[prior.ind][[i]], list(p=mcmc.out[[c]][,i]))
+        m[,i] <- eval(prior.fn$qprior[prior.ind][[i]], list(p=mcmc.out[[c]]$mcmc[,i]))
       }
     }
-      colnames(m) <- pname[prior.ind]
-      mcmc.out[[c]] <- m
+    colnames(m) <- pname[prior.ind]
+    mcmc.list.tmp[[c]] <- m
+    
+    jvar.list[[c]] <- sqrt(mcmc.out[[c]]$jump@history[nrow(mcmc.out[[c]]$jump@history),])
```

Comment:
I don't know much about S3 vs. S4 classes, but from poking around a bit, people seem to recommend sticking to S3 classes unless there is a very good reason to use S4. I have no idea how the `jump` object is designed, but it may be worth thinking about tweaking it so we can use the older, simpler class system. Again, just a thought -- if this works for now, it's totally fine to keep it.